app/server.go

```diff
@@ -86,32 +88,61 @@ func RunServer(c *Config) error {
 
 	iamManager := identities.NewManager(crypto, v)
 
-	resourceService := resource.NewService(resourceRepository)
-	providerService := provider.NewService(
-		logger,
-		v,
-		providerRepository,
-		resourceService,
-		providerClients,
-	)
-	policyService := policy.NewService(
-		v,
-		policyRepository,
-		resourceService,
-		providerService,
-		iamManager,
+	auditRepository := audit_repos.NewPostgresRepository(db)
+	auditLogger := audit.New(
+		audit.WithRepository(auditRepository),
+		audit.WithAppDetails(audit.AppDetails{
+			Name:    "guardian",
+			Version: Version,
+		}),
+		audit.WithTrackIDExtractor(func(ctx context.Context) string {
+			if md, ok := metadata.FromIncomingContext(ctx); ok {
+				if rawTraceID := md.Get(c.AuditLogTraceIDHeaderKey); len(rawTraceID) > 0 {
+					return rawTraceID[0]
+				}
+			}
+			return ""
```

Comment:
Does it mean if TraceID not found in the header, we still audit it but with empty TraceID?

Reply:
we should generate a TraceID. Need to discuss the trace ID convention first. Marking it as TODO